src/Appwrite/Auth/OAuth2/Disqus.php

```diff
@@ -0,0 +1,134 @@
+<?php
+
+namespace Appwrite\Auth\OAuth2;
+
+use Appwrite\Auth\OAuth2;
+
+// Reference Material
+// https://disqus.com/api/docs/auth/
+
+class Disqus extends OAuth2
+{
+    private string $endpoint = 'https://disqus.com/api/';
+    protected array $user = [];
+    protected array $tokens = [];
+    protected array $scopes = [
+        'read',
+        'email',
+    ];
+
+    public function getName(): string
+    {
+        return 'disqus';
+    }
+
+    public function getLoginURL(): string
```

Comment:
```suggestion
    /**
     * @return string
     */
    public function getLoginURL(): string
```

All our functions need to have a doc block to indicate their return types and any parameters that they accept.
Please refer to https://github.com/appwrite/appwrite/blob/master/src/Appwrite/Auth/OAuth2/Linkedin.php

Reply:
Understood. That being said, I guess we need to add comments in this snippet too [`add-oauth2-provider.md`](https://github.com/appwrite/appwrite/blob/master/docs/tutorials/add-oauth2-provider.md#23-add-provider-class) too. Cause, most of the contributors who would work on adding an OAuth provider would copy this skeleton code